Source/carthage/Outdated.swift

```diff
@@ -54,13 +82,18 @@ public struct OutdatedCommand: CommandProtocol {
 
 				if !outdatedDependencies.isEmpty {
 					carthage.println(formatting.path("The following dependencies are outdated:"))
-					for (project, current, updated, latest) in outdatedDependencies {
+
+					for (project, current, applicable, latest) in outdatedDependencies {
 						if options.outputXcodeWarnings {
-							carthage.println("warning: \(formatting.projectName(project.name)) is out of date (\(current) -> \(updated)) (Latest: \(latest))")
+							carthage.println("warning: \(formatting.projectName(project.name)) is out of date (\(current) -> \(applicable)) (Latest: \(latest))")
 						} else {
-							carthage.println(formatting.projectName(project.name) + " \(current) -> \(updated) (Latest: \(latest))")
+							let update = UpdateType(currentVersion: current, applicableVersion: applicable, latestVersion: latest)
+							let style = formatting[update]
+							let versionSummary = "\(style(current.description)) -> \(style(applicable.description)) (Latest: \(latest))"
+							carthage.println(formatting.projectName(project.name) + " " + versionSummary)
 						}
 					}
+					formatting.legendForOutdatedCommand.map(carthage.println)
```

Comment:
I don't think the legend should be prevented on every invocation. IMO this information should be included in `carthage help outdated` instead.

Reply:
I'm unsure how to do that. The simple solution I see is to add the text to
`public let function = "Check for compatible updates to the project's dependencies"`, then it will be printed at the top. But that seems wrong to me.

I'd like it much better to include the explanation to the help text of the color option, e.g.:

```
[--color (color)]
	whether to apply color and terminal formatting (one of 'auto', 'always', or 'never')
	
	The color indicates what happens when you run `carthage update`
	<green>	- Will be updated to the newest version
	<yellow>	- Will be updated, but not to the newest version
	<red>	- Will not be updated because of the specified version in Cartfile
```

However, I have no Idea how to implement this since the color option help is generated by `ColorOptions.evaluate(mode)` and I don't see a way to add extra text to the help of `ColorOptions`...

Reply:
I think it'd be fine to change it to:

```swift
public static func evaluate(_ mode: CommandMode, additionalUsage: String? = nil) -> Result<ColorOptions, CommandantError<CarthageError>>
```

Then you could pass the extra information for `outdated` and have it appended.

Reply:
Do you mean actually changing the method in [Commandant.OptionsProtocol](https://github.com/Carthage/Commandant/blob/master/Sources/Commandant/Option.swift#L44) or just adding an override with the extra parameter to `Formatting`?